### ISSUE TITLE
`mx fetch-jdk` fails to get labsjdk-ce-22

### ISSUE THREAD

**Issue by zakkak** [oracle/graal#7425]
**Describe the issue**
`mx fetch-jdk` fails to get labsjdk-ce-22. 
https://github.com/graalvm/labs-openjdk-22 appears to be private.

**Steps to reproduce the issue**
```
❯ mx fetch-jdk --to ~/jvms --java-distribution labsjdk-ce-22
Install labsjdk-ce-22-ce-22+13-jvmci-b01 to /home/zakkak/jvms/labsjdk-ce-22-ce-22+13-jvmci-b01? [Yn]: 
Fetching labsjdk-ce-22-ce-22+13-jvmci-b01 archive from https://github.com/graalvm/labs-openjdk-22/releases/download/ce-22%2B13-jvmci-b01/labsjdk-ce-22%2B13-jvmci-b01-linux-amd64.tar.gz...
Error retrieving https://github.com/graalvm/labs-openjdk-22/releases/download/ce-22%2B13-jvmci-b01/labsjdk-ce-22%2B13-jvmci-b01-linux-amd64.tar.gz.sha1: HTTP Error 404: Not Found
```

**Describe GraalVM and your environment:**
 - GraalVM version (latest snapshot builds can be found [here](https://github.com/graalvm/graalvm-ce-dev-builds/releases)), or commit id if built from source: **5623fe69c3f2c4f907598b9bbf99ebe3a63d6359**

cc @dougxc @marwan-hallaoui

**Comment by dougxc** [comment]
We've adopted the OpenJDK convention where the latest labsjdk repo does not include a version (just like https://github.com/openjdk/jdk does not include a version). So, what you're looking for is now at https://github.com/graalvm/labs-openjdk.

We still have to fix `mx fetch-jdk` for this change I guess (cc @zapster ).

**Comment by zapster** [comment]
Thanks for the report @zakkak. I'll have a look.

**Comment by zapster** [comment]
The problem is actually in [mx ](https://github.com/graalvm/mx/blob/master/jdk-binaries.json), which does not yet know about the https://github.com/graalvm/labs-openjdk repo. However, to keep friction low when doing the JDK 22 fork-off, i.e., when `labsjdk-ce-22` actually moves into the https://github.com/graalvm/labs-openjdk-22 repository, we will rename the latest labsjdk from `labsjdk-ce-22`  to `labsjdk-ce-latest`. That way, the jdk binary location does not need to change when the OpenJDK bumps the version from 22 to 23 on master.

**Comment by zapster** [comment]
`mx` support for `labsjdk-ce-latest` as landed in [aa775d2](https://github.com/graalvm/mx/commit/aa775d2778338563f25830e71b7736298e6770e7).
The `labsjdk-*-22` to `labsjdk-*-latest` rename happens in #7430.
